addons/website_forum/models/forum.py

```diff
@@ -509,8 +509,8 @@ def _update_content(self, content, forum_id):
         forum = self.env['forum.forum'].browse(forum_id)
         if content and self.env.user.karma < forum.karma_dofollow:
             for match in re.findall(r'<a\s.*href=".*?">', content):
-                match = re.escape(match)  # replace parenthesis or special char in regex
-                content = re.sub(match, match[:3] + 'rel="nofollow" ' + match[3:], content)
+                escaped_match = re.escape(match)  # replace parenthesis or special char in regex
+                content = re.sub(escaped_match, match[:3] + 'rel="nofollow" ' + match[3:], content)
```

Comment:
I think I see what you mean, if we already have a "rel" inside the link tag, this does not work properly.

Another more simple way to do it would be to catch the URL and rebuild the tag completely.

Something like:
```py
escaped_match = re.escape(match)  # replace parenthesis or special char in regex
url_match = re.match(r'^.*href="(.*)".*', match) # extracting the link allows to rebuild a clean link tag
url = url_match.group(1)
content = re.sub(escaped_match, f'<a rel="nofollow" href="{url}">', content)
```

That way we ensure a correct "nofollow" ref with the URL and a cleaned tag.

Can you give it a try?
(I did not test the regex, just wrote it directly, will maybe need a couple tweaks).